python/lsst/pipe/tasks/insertFakes.py

```diff
@@ -835,11 +861,23 @@ def _generateGSObjectsFromCatalog(self, exposure, fakeCat, galCheckVal, starChec
                 gal = bulge*row['bulge_disk_flux_ratio'] + disk
                 gal = gal.withFlux(flux)
 
-                yield skyCoord, gal
+                yield skyCoord, gal, "gal"
```

Comment:
What is the added string for?

Reply:
The galsim object for a trail needs to be updated inside of the `_add_fake_sources`, so this is just passing the source type to that function. There's nothing done if it's not `"trail"`, but I added strings for each just for consistency.

Reply:
Can that not be done within the generateObjects function?

Reply:
It could, but it would duplicate all the code to obtain the `gsWCS` on line 85. I suppose the relevant code could be put in a function and used both here and in generateObjects. Would that be preferable?

Reply:
I don't think it would be a huge amount of code duplication, I think it should just be lines 84 and 85. If we forsee needing it in more places as we develop the code further than we could pass it to the function. Otherwise I think duplicating those two lines here is preferable.

Reply:
Hmm, you're definitely right. I thought there was way more code to copy... I'll change that and remove the string arguments